packages/vitest/src/node/pools/rpc.ts

```diff
@@ -20,8 +26,33 @@ export function createMethodsRPC(project: WorkspaceProject): RuntimeRPC {
       const r = await project.vitenode.transformRequest(id)
       return r?.map as RawSourceMap | undefined
     },
-    fetch(id, transformMode) {
-      return project.vitenode.fetchModule(id, transformMode)
+    async fetch(id, transformMode) {
+      const result = await project.vitenode.fetchModule(id, transformMode)
+      if (result.externalize)
+        return result
+      if (!result.code && 'id' in result)
+        return result
+
+      if (!result.code) {
+        console.error(result)
+        return { code: `throw new Error('What is going on?')` }
+      }
+      const code = result.code
+      const dir = join(tmpdir(), transformMode)
```

Comment:
instead of tmpdir, we can use a `cache.dir`

Reply:
our current caching strategy is all over the place, and in dire need of a refactor